apps/server/src/shared/domain/entity/lesson.entity.ts

```diff
@@ -9,6 +9,19 @@ export interface ILessonProperties {
 	hidden?: boolean;
 	course: Course;
 	position?: number;
+	contents: IComponentProperties[] | [];
+}
+
+export interface IContentProperties {
+	content: {
+		text: string;
+	};
+}
+
+export interface IComponentProperties extends IContentProperties {
+	title: string;
+	hidden?: boolean;
```

Comment:
hideen should not be optional
missing user propriety

Reply:
naming IComponentTextProperties

Reply:
hide lesson after copy, but not content

Reply:
the name could contain Lesson in there too, since we use this elsewhere, is easier to understand what kind of component. ILessonComponent for example